.github/workflows/python-package.yml

```diff
@@ -37,33 +56,42 @@ jobs:
               uses: actions/checkout@v4
               with:
```

Comment:
Just a suggestion:
```
with:
    fetch-depth: 0
```
It turns out this doesn't do what you might think. `fetch-depth = 0` means "fetch all commits". When in fact we want "fetch the latest commit", which is the *default* behavior of `actions/checkout`. So the `with:` clause in that step isn't needed at all. If you want to remove these in this PR, go ahead, but it's also something that could be done later.